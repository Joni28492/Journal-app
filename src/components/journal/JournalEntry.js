import React from 'react';
import moment from 'moment';
import { useDispatch } from 'react-redux';
import { activeNote } from '../../actions/notes';

export const JournalEntry = ({id, title, date,body, url }) => {
  
    const noteDate = moment(date);
    const dispatch = useDispatch();

    const handleEntryClick = () =>{
        dispatch(activeNote(id, {title, date,body, url}));
    }

    return (
        <div className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
            onClick={handleEntryClick}
        >

            {   url && //condicion para saber si hay imagen
                <div className="journal__entry-picture"
                style={{
                    backgroundSize: 'cover',//aqui sin guiones
                    backgroundImage: `url(${url})`,//https://cdna.artstation.com/p/assets/images/images/006/458/524/large/joni-fernandez-03.jpg?1498739545
                }}
                ></div>
            }

            <div className="journal__entry-body">
                <p className="journal__entry-title">{title}</p>
                <p className="journal__entry-content">{body}</p>
            </div>

            <div className="journal__entry-date-box">
                <span>{noteDate.format('dddd')}</span>
                <h4>{noteDate.format('Do')}</h4>
            </div>
        </div>
    )
}
