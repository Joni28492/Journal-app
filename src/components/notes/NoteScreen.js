

import React, { useEffect, useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { activeNote, startDeleting } from '../../actions/notes';
import { useForm } from '../../hooks/useForm';
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {

    const {active:note} = useSelector(state => state.notes);
    const [formValues, handleInputChange, reset] =useForm(note);
    const {body, title, id}=formValues;
    const dispatch = useDispatch();

    //almacena una var mutable q no varia todo el comp si cambia
    const activeId =useRef(note.id)

    useEffect(() => {
        if (note.id !== activeId.current) {
            reset(note);
            activeId.current=note.id;
        }
    }, [note, reset])


    useEffect(() => {//la accion remplaza el id
      dispatch(activeNote(formValues.id, {...formValues}));
    }, [formValues, dispatch]);


    const handleDelete = () =>{
        dispatch(startDeleting(id));
    }
    

    return (
        <div className="notes__main-content">
            <NotesAppBar />

            <div className="notes__content">
                <input 
                    type="text" 
                    placeholder="Some awesome title"
                    className="notes__title-input" 
                    autoComplete="off"
                    value={title}
                    onChange={handleInputChange}
                    name='title'
                />

                <textarea
                    placeholder="What happened today?"
                    className="notes__textarea"
                    value={body}
                    onChange={handleInputChange}
                    name='body'
                ></textarea>

                {   (note.url) &&
                    (<div className="notes__image">
                        <img 
                            src={note.url}
                            alt="imagen"
                        />
                    </div>)
                }
            </div>


            <button onClick={handleDelete} className="btn btn-danger">
                Delete
            </button>
        </div>
    )
}
