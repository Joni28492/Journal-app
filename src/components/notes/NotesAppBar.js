import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startSaveNote, startUploading } from '../../actions/notes'

export const NotesAppBar = () => {

    const {active} = useSelector(state => state.notes)
    const dispatch = useDispatch()

    const handleSave = () =>{
        dispatch(startSaveNote(active))
    }

    const handlePictureClick = () =>{
        document.querySelector('#fileSelector').click();
    }

    const handleFileChange = (e) =>{
        const file = e.target.files[0];
        //controlamos el cancel
        if (file) {
            dispatch(startUploading(file))
        }
    }

    return (
        <div className="notes__appbar">
            <span>30 Marzo 2021</span>

            <input type="file" style={{display: 'none'}} 
                onChange={handleFileChange} id='fileSelector'
                name="file"
            />

            <div>
                <button className="btn" onClick={handlePictureClick} >Picture</button>
                <button className="btn" onClick={handleSave}>save</button>
            </div>
        </div>
    )
}
