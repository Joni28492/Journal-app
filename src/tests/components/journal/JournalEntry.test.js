
import '@testing-library/jest-dom';
import React from 'react';
import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { JournalEntry } from "../../../components/journal/JournalEntry";
import { activeNote } from '../../../actions/notes';




const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState= {
    auth: {
        uid: '1',//Da igual el uid
        name: 'Joni'
    },
    ui: {
        loading: false,
        msgError: null   
    },
    notes: {
        active: null,
        notes: [],
    }
};
let store = mockStore(initState);
store.dispatch=jest.fn();


const nota = {
    id: 10,
    date: 0,
    title: 'hola',
    body: 'mundo',
    url: 'https://algunlugar.com/foto.jpg'
};
const wrapper = mount(
    <Provider store={store}>
            <JournalEntry {...nota}/>
    </Provider>
);



describe('Pruebas en <JournalEntry />', () => {
    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });


    test('debe de activar la nota', () => {
        wrapper.find('.journal__entry').prop('onClick')();
        // expect(store.dispatch).toHaveBeenCalledWith();
        expect(store.dispatch).toHaveBeenCalledWith(
            activeNote(nota.id, {...nota})
        );
    })
    
    
})
