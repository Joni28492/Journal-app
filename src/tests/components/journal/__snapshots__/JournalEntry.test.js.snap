// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <JournalEntry /> debe mostrarse correctamente 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://algunlugar.com/foto.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      hola
    </p>
    <p
      className="journal__entry-content"
    >
      mundo
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Thursday
    </span>
    <h4>
      1st
    </h4>
  </div>
</div>
`;
